### PR TITLE
DM-47821: Add utility functions for consistent multiband plot colors

### PR DESCRIPTION
**Pull request by isullivan**
## Checklist

- [x ] ran Jenkins
- [ x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Comment by timj** [comment]
Please use underscores in the function names. This package does not use camel case.

File: python/lsst/utils/plotting/figures.py [hunk 1]
```diff
@@ -55,3 +60,96 @@ def make_figure(**kwargs: Any) -> Figure:
...
+
+    Notes
+    -----
+    From https://rtn-045.lsst.io/#colorblind-friendly-plots
+
+    Parameters
+    ----------
+    background : `str`, optional
+        Selection of colors intended for a white or a black background.
+        Allowed values are 'white' or 'black'
```
Comment (mfisherlevine) [inline python/lsst/utils/plotting/figures.py:76]: I don't have _strong_ feelings on this, but if you used a boolean kwarg like `dark_mode` or `black_background` then people wouldn't need to know/remember the magic values, and you wouldn't have to validate the input either.

File: python/lsst/utils/plotting/figures.py [hunk 2]
```diff
@@ -55,3 +60,96 @@ def make_figure(**kwargs: Any) -> Figure:
...
+        "g": "#49be61",
+        "r": "#c61c00",
+        "i": "#ffc200",
+        "z": "#f341a2",
+        "y": "#5d0000",
+    }
+    plot_filter_colors_black_background = {
+        "u": "#3eb7ff",
+        "g": "#30c39f",
+        "r": "#FF7E00",
```
Comment (mfisherlevine) [inline python/lsst/utils/plotting/figures.py:99]: I don't think it matters, but there's random case inconsistency which makes me uncomfortable 😄

File: tests/test_matplotlib_figures.py [hunk 3]
```diff
@@ -45,7 +50,30 @@ def testMakeFigure(self):
...
+                    linestyle=linestyles[band],
+                )
+            for band in bands:
+                ax.plot(
+                    [0, 1],
+                    [0, 1],
+                    c=alt_colors[band],
+                    marker=symbols[band],
+                    linestyle=linestyles[band],
+                )
```
Comment (mfisherlevine) [inline tests/test_matplotlib_figures.py:75]: If you added a loop over the background colors then you'd hit every combination and get full coverage and could only need one series on the plot, _i.e._

```
for background in 'white', 'black':
    colors = get_multiband_plot_colors(background=background)
    symbols = get_multiband_plot_symbols(background=background)
    linestyles = get_multiband_plot_linestyles(background=background)

    ax = fig.add_subplot(111)
    ax.plot([0, 1], [0, 1], "r-")
    for band in bands:
        ax.plot([0, 1],[0, 1], c=colors[band], marker=symbols[band], linestyle=linestyles[band])
```

File: python/lsst/utils/plotting/figures.py [hunk 4]
```diff
@@ -55,3 +60,96 @@ def make_figure(**kwargs: Any) -> Figure:
...
+        "i": "#2af5ff",
+        "z": "#a7f9c1",
+        "y": "#fdc900",
+    }
+    if background == "white":
+        return plot_filter_colors_white_background
+    elif background == "black":
+        return plot_filter_colors_black_background
+    else:
+        raise ValueError("'background' must be either 'white' or 'balck'")
```
Comment (mfisherlevine) [inline python/lsst/utils/plotting/figures.py:109]: `balck` → `black` if you don't move to a bool.

File: doc/changes/DM-47821.feature.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Added ``lsst.utils.plotting.getMultibandPlotColors``, ``lsst.utils.plotting.getMultibandPlotSymbols``, and ``lsst.utils.plotting.getMultibandPlotLinestyles`` utilities for making consistent multi-band plots.
```
Comment (mfisherlevine) [inline doc/changes/DM-47821.feature.rst:1]: Need to make these into camel case now that you've made Tim's changes elsewhere.

File: python/lsst/utils/plotting/figures.py (hunk 2 above)
Comment (isullivan) [reply python/lsst/utils/plotting/figures.py:99]: Fair point! I just copied and pasted from the technote, but I'll fix it.

**Pull request merged**